src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java

```diff
@@ -0,0 +1,412 @@
+        assertNotNull(refreshJob.getLog());
+    }
+
+    @Test
+    public void shouldFailProcessRefreshRoleAssignmentsWithFailedUsersAndWithOutJobID() throws Exception {
+        logger.info(" Refresh Job with optional Users and without mandatory jobId as a param");
+        mockMvc.perform(post(URL)
+                .contentType(CONTENT_TYPE)
+                .headers(getHttpHeaders())
+                .content(mapper.writeValueAsBytes(IntTestDataBuilder.buildUserRequest())))
+                .andExpect(status().is(415))
+                .andReturn();
+    }
+
+    @Test
+    public void shouldFailProcessRefreshRoleAssignmentsWithEmptyJobID() throws Exception {
+        logger.info(" Refresh Job with optional Users and without mandatory jobId as a param");
+        mockMvc.perform(post(URL)
+                .contentType(CONTENT_TYPE)
+                .headers(getHttpHeaders())
+                .param("jobId", ""))
+                .andExpect(status().is(415))
+                .andReturn();
+    }
+
+    @Test
+    public void shouldFailProcessRefreshRoleAssignmentsWithInvalidJobID() throws Exception {
+        logger.info(" Refresh Job with optional Users and without mandatory jobId as a param");
+        mockMvc.perform(post(URL)
+                .contentType(CONTENT_TYPE)
+                .headers(getHttpHeaders())
+                .param("jobId", "abc"))
+                .andExpect(status().is(415))
+                .andReturn();
+    }
+
+    @Test
+    public void shouldFailProcessRefreshRoleAssignmentsWithOutJobID() throws Exception {
+        logger.info(" Refresh Job with optional Users and without mandatory jobId as a param");
+        mockMvc.perform(post(URL)
+                .contentType(CONTENT_TYPE)
+                .headers(getHttpHeaders()))
+                .andExpect(status().is(400))
+                .andReturn();
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_refresh_jobs.sql"})
+    public void shouldProcessRefreshRoleAssignmentsWithJobIdToComplete_retryFail() throws Exception {
+        logger.info(" RefreshJob record With Only JobId to process fail");
+        Long jobId = 1L;
+        RefreshJobEntity refreshJob = getRecordsFromRefreshJobTable(jobId);
+        assertEquals("NEW", refreshJob.getStatus());
+
+        doThrow(RuntimeException.class).when(crdFeignClient).getCaseworkerDetailsByServiceName(
+                anyString(), anyInt(), anyInt(), anyString(), anyString());
+
+        mockMvc.perform(post(URL)
+                .contentType(JSON_CONTENT_TYPE)
+                .headers(getHttpHeaders())
+                .param("jobId", jobId.toString()))
+                .andExpect(status().is(202))
+                .andReturn();
+
+        refreshJob = getRecordsFromRefreshJobTable(jobId);
+        assertEquals("NEW", refreshJob.getStatus());// failed process should change the status to IN-PROGRESS
```

Comment:
It is not making it in-progress so that next pod instance of refresh batch job can pick up and resend the same entry. This scenario would happen in case of recoverable errors and next batch job pod will trigger after 30 mins.